pkg/bloomcompactor/bloomcompactor.go

```diff
@@ -106,141 +173,263 @@ func (c *Compactor) stopping(_ error) error {
+
+	blocks := bloomshipper.Block{
+		BlockRef: bloomshipper.BlockRef{
+			Ref: bloomshipper.Ref{
+				TenantID:       series.tenant,
+				TableName:      series.tableName,
+				MinFingerprint: uint64(series.fingerPrint), //TODO will change once we compact multiple blooms into a block
+				MaxFingerprint: uint64(series.fingerPrint),
+				StartTimestamp: series.from.Unix(),
+				EndTimestamp:   series.through.Unix(),
+				Checksum:       binary.BigEndian.Uint32(checksum),
+			},
+			IndexPath: series.indexPath, // For reviewer, this is the same filepath with the original TSDBindex for this series?
+			BlockPath: objectStoreDst,
+		},
+		Data: blockFile,
 	}
-	// BuildFrom closes itself
-	err = builder.BuildFrom(v1.NewSliceIter[v1.SeriesWithBloom](seriesList))
+
+	return blocks, nil
+}
+
+func listSeries(ctx context.Context, objectClient storeClient) ([]Series, error) {
+	// Returns all the TSDB files, including subdirectories
+	prefix := "index/"
+	indices, _, err := objectClient.object.List(ctx, prefix, "")
+
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	// TODO Ask Owen, shall we expose a method to expose these paths on BlockWriter?
-	indexPath := filepath.Join(dst, "series")
-	bloomPath := filepath.Join(dst, "bloom")
+	var result []Series
+
+	for _, index := range indices {
+		s := strings.Split(index.Key, "/")
+
+		if len(s) > 3 {
+			tableName := s[1]
+
+			if !strings.HasPrefix(tableName, "loki_") || strings.Contains(tableName, "backup") {
+				continue
+			}
+
+			userID := s[2]
+			_, err := strconv.Atoi(userID)
+			if err != nil {
+				continue
+			}
 
-	blockRef := bloomshipper.BlockRef{
-		IndexPath: indexPath,
-		BlockPath: bloomPath,
+			result = append(result, Series{tableName: tableName, tenant: userID, indexPath: index.Key})
+		}
 	}
+	return result, nil
+}
 
-	blocks := []bloomshipper.Block{
-		{
-			BlockRef: blockRef,
+func createLocalFileName(series Series) string {
+	return fmt.Sprintf("bloomBlock-%s-%s-%s-%s-%s-%s", series.tableName, series.tenant, series.fingerPrint, series.fingerPrint, series.from, series.through)
+}
+func createObjStorageFileName(series Series, checksum string) string {
+	// TODO fix series_fp repeating when creating multi-series bloom blocks
+	// From design doc, naming should be something like
+	//`bloom/<period>/<tenant>/blooms/<start_fp>-<end_fp>/<start_ts>-<end_ts>-<checksum>`
+	fileName := fmt.Sprintf("blooms/%s-%s/%s-%s-%s", series.fingerPrint, series.fingerPrint, series.from, series.through, checksum)
+	return strings.Join([]string{"bloom", series.tableName, series.tenant, fileName}, "/")
+}
 
-			// TODO point to the data to be read
-			Data: nil,
+func CompactNewChunks(ctx context.Context, series Series, bloomShipperClient bloomshipper.Client) (err error) {
+	// Create a bloom for this series
+	bloomForChks := v1.SeriesWithBloom{
+		Series: &v1.Series{
+			Fingerprint: series.fingerPrint,
+		},
+		Bloom: &v1.Bloom{
+			ScalableBloomFilter: *filter.NewDefaultScalableBloomFilter(0.01),
```

Comment:
nit: let's move fpRate(`0.01`) to the config with visibility `advanced` ... just in case we decide to tune it in the future to avoid the necessity to rebuild it

Reply:
I'll move to a constant for now to make it more clear.
But i'm not sure what you mean by "to avoid the necessity to rebuild it" because changing falsePositive rate will require bloomFilter to be re-created. We can't change it in runtime. I've checked a while ago with @paul1r that it's a decision to configure it, i don't think we need it in config.

Reply:
it's not critical, so np if we do not have it in the config.

>to avoid the necessity to rebuild it

`it` here is loki image ) sorry for the confusion 

I meant to add this field to the loki config (not runtime) so we could tune it and restart the compactor and it would use the new fp rate...